R/render_as_i_html.R

```diff
@@ -55,94 +55,80 @@ render_as_ihtml <- function(data, id) {
-
-      lang_defs <- reactable::reactableLang()
-
-    } else {
-
-      lang_defs <-
-        reactable::reactableLang(
-          sortLabel = locale_data[["sort_label_text"]],
-          filterPlaceholder = "",
-          filterLabel = locale_data[["filter_label_text"]],
-          searchPlaceholder = locale_data[["search_placeholder_text"]],
-          searchLabel = locale_data[["search_placeholder_text"]],
-          noData = locale_data[["no_table_data_text"]],
-          pageNext = locale_data[["page_next_text"]],
-          pagePrevious = locale_data[["page_previous_text"]],
-          pageNumbers = locale_data[["page_numbers_text"]],
-          pageInfo = gsub("\\\\u2013", "\u2013", locale_data[["page_info_text"]]),
-          pageSizeOptions = locale_data[["page_size_options_text"]],
-          pageNextLabel = locale_data[["page_next_label_text"]],
-          pagePreviousLabel = locale_data[["page_previous_label_text"]],
-          pageNumberLabel = locale_data[["page_number_label_text"]],
-          pageJumpLabel = locale_data[["page_jump_label_text"]],
-          pageSizeOptionsLabel = locale_data[["page_size_options_label_text"]],
-          groupExpandLabel = "Toggle group",
-          detailsExpandLabel = "Toggle details",
-          selectAllRowsLabel = "Select all rows",
-          selectAllSubRowsLabel = "Select all rows in group",
-          selectRowLabel = "Select row"
-        )
-    }
+    rowname_label <- stub_label
+    groupname_label <- NULL
   }
 
+
   # Obtain the underlying data table (including group rows)
   data_tbl0 <- dt_data_get(data = data)
 
-  #
   # Only preserve columns that are not hidden (group cols will be added later)
-  #
-
   data_tbl_vars <- dt_boxhead_get_vars_default(data = data)
   data_tbl <- data_tbl0[, data_tbl_vars, drop = FALSE]
 
   #nocov start
-
-  # Stop function if there are no visible columns
   if (ncol(data_tbl) < 1) {
-
+    # Stop function if there are no visible columns
     cli::cli_abort(c(
       "When displaying an interactive gt table, there must be at least one visible column.",
       "*" = "Check that the input data table has at least one column,",
       "*" = "Failing that, look at whether all columns have been inadvertently hidden."
     ))
   }
-
   #nocov end
 
-  rownames_to_stub <- stub_rownames_has_column(data)
   # use of special .rownames doesn't work.
   # Workaround https://github.com/glin/reactable/issues/378
   # rstudio/gt#1702
   if (rownames_to_stub) {
     rowname_col <- dt_boxhead_get_var_stub(data)
     if (length(rowname_col) == 1) {
       # avoid base R error when setting duplicate row names.
-      attr(data_tbl, "row.names") <-  as.character(data$`_data`[[rowname_col]])
+      row_names <-  as.character(data$`_data`[[rowname_col]])
+      # Convert to NA string to avoid wrong output.
+      # TODO figure out if there is a way to get the sub_missing value.
+      # With data$`_substitutions`
+      row_names <- dplyr::coalesce(row_names, "NA")
```

Comment:
```suggestion
      row_names <- dplyr::coalesce(row_names, "")
```

For consistency with `reactable()` in general, better to display nothing and we can fix this in the future with #1759